test/worker_test.py

```diff
@@ -1042,6 +1042,62 @@ def test_stop_getting_new_work_assistant_keep_alive(self):
         self._test_stop_getting_new_work(keep_alive=True, assistant=True)
 
 
+class WorkerLockTest(LuigiTestCase):
+
+    class LockTask(RunOnceTask):
+
+        name = luigi.Parameter()
+        delay1 = luigi.FloatParameter()
+        delay2 = luigi.FloatParameter()
+        use_locking = luigi.BoolParameter()
+        log_file = luigi.Parameter()
+
+        def write_line(self, line):
+            with open(self.log_file, "a") as f:
+                f.write(line + "\n")
+
+        def run(self):
+            time.sleep(self.delay1)
+            if self.use_locking:
+                self.worker_lock.acquire()
+            self.write_line("start " + self.name)
+            time.sleep(self.delay2)
+            self.write_line("end " + self.name)
+            if self.use_locking:
+                self.worker_lock.release()
```

Comment:
Suppose someone uses this feature, but doesn't release the lock. That wouldn't be good. Perhaps provide some documentation on the feature and possible use cases.

Reply:
Yup, I will add a section to the "Luigi Patterns" once the tests look fine.

Reply:
Also, I would encourage to use the lock context guard in the docs to avoid trouble for users forgetting to release the lock in most cases.